### ISSUE TITLE
Is it possible to implement Tensorboard in Darknet?

### ISSUE THREAD

**Issue by RobinV760** [AlexeyAB/darknet#6678]
I am training a custom detection with YOLOv4 using this repo. 

I have recently become familiar with TensorBoard to keep track of training processes, and I wanted to implement it here. However, I could not find the appropriate place to implement TensorBoard. 

Does anyone have experience with implementing it in Darknet? Or is it not possible because in the end Darknet is written in C?

**Comment by a-rich** [comment]
Run training like this:

`./darknet detector train path/to/yolo.data path/to/yolov4.cfg path/to/starting.weights -map -dont_show -gpus 0 -mjpeg_port 8090 > log.txt`

Then, if you're training locally, you can watch training loss / validation accuracy on port 8090...

If you're training remotely, bind port 8090 via ssh:
`ssh -L 3000:127.0.0.1:8090 your_server_user@your_server_IP`

**Comment by RobinV760** [comment]
Thanks for your comment! I am indeed training remotely (via AWS EC2). I will try to bind port via ssh. 

Would you mind specifying what the 3000:127.0.0.1:8090 stands for? Is it always this code, or should/can it be altered? Also, what is your_server_user@your_server_IP? Is that the public DNS from AWS? Like ubuntu@... ? or something locally?

I'm starting to get familiar with AWS and SSH but I haven't used this before. If you have any tips on documentation that's also appreciated :)

**Comment by a-rich** [comment]
> Would you mind specifying what the 3000:127.0.0.1:8090 stands for?
<local_port>:<IP_on_remote_server>:<remote_port>
Because you ran darknet with `-mjpeg_port 8090`, the `<remote_port>` part of the command must also be `8090`...idk what degree of success you'll have running `-mjpeg_port` with a different port number so I would leave that as is.

`127.0.0.1` is the IP address of local host...so what this is saying is that, from your remote server, you want to bind port 8090 of that machine itself rather than some other machine on the same network.

The `<local_port>` can be anything, though I would stick to port numbers >= 3000 so you don't have conflict with the dedicated ports that are < 3000.

> If I run this command (from the .ssh folder on my local computer)
It's not necessary to run this from your `~/.ssh` folder

> I get the message 'permission denied (publickey)'. But I don't have any problems logging into the remote server with ssh
Paste your successful ssh login and your unsuccessful ssh login with port binding -- it shouldn't make a difference...

You can also just go to the public HTTP address of your EC2 instance...something like this (obviously change the IP address numbers and make sure you have the correct server location):
`http://ec2-35-160-228-91.us-west-2.compute.amazonaws.com:8090`
